keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/ScalarDoubleTensorTest.java

```diff
@@ -229,4 +229,29 @@ public void doesKeepRankOnLTEq() {
         DoubleTensor value = DoubleTensor.create(new double[]{1}, 1, 1, 1);
         assertEquals(3, value.lessThanOrEqual(2).getRank());
     }
+
+    @Test
+    public void doesMatrixMultiplyWhenRank2() {
+        DoubleTensor lengthOne = new ScalarDoubleTensor(2).reshape(1, 1);
+        DoubleTensor matrix = DoubleTensor.create(1, 2, 3, 4).reshape(2, 2);
+        DoubleTensor result = lengthOne.matrixMultiply(matrix);
+        DoubleTensor expected = DoubleTensor.create(2, 4, 6, 8).reshape(2, 2);
+        assertEquals(expected, result);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void doesThrowOnMatrixMultiplyWhenRank0() {
+        DoubleTensor lengthOne = new ScalarDoubleTensor(2);
+        DoubleTensor matrix = DoubleTensor.create(1, 2, 3, 4).reshape(2, 2);
+        lengthOne.matrixMultiply(matrix);
+    }
+
+    @Test
+    public void doesTensorMultiply() {
+        DoubleTensor lengthOne = new ScalarDoubleTensor(2).reshape(1);
```

Comment:
what do you expect to happen when you remove the `reshape`?

Currently:

    java.lang.ArrayIndexOutOfBoundsException: 0

	at io.improbable.keanu.tensor.TensorShape.getTensorMultiplyResultShape(TensorShape.java:261)
	at io.improbable.keanu.tensor.dbl.ScalarDoubleTensor.tensorMultiply(ScalarDoubleTensor.java:171)
	at io.improbable.keanu.tensor.dbl.ScalarDoubleTensorTest.doesTensorMultiply(ScalarDoubleTensorTest.java:253)

Reply:
Added a check for validating that the specified dimensions actually exists.